File: menu-scanner-backend/src/main/java/com/emenu/features/subscription/dto/response/SubscriptionHistoryResponse.java
package com.emenu.features.subscription.dto.response;

import com.emenu.enums.payment.PaymentMethod;
import com.emenu.enums.sub_scription.SubscriptionPaymentStatus;
import com.emenu.enums.sub_scription.SubscriptionPaymentType;
import com.emenu.enums.sub_scription.SubscriptionPlanDurationType;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Data
public class SubscriptionHistoryResponse {

    private UUID subscriptionId;
    private UUID businessId;
    private String businessName;
    private String logoBusinessUrl;
    private UUID planId;
    private String planName;
    private BigDecimal planPrice;
    private SubscriptionPlanDurationType planDurationType;
    private LocalDate startDate;
    private LocalDate endDate;
    private Boolean autoRenew;
    private String status;
    private Long daysRemaining;
    private String paymentStatus;
    private BigDecimal totalPaid;
    private PaymentItem payment;

    @Data
    public static class PaymentItem {
        private UUID paymentId;
        private BigDecimal amount;
        private PaymentMethod paymentMethod;
        private SubscriptionPaymentType paymentType;
        private SubscriptionPaymentStatus status;
        private String referenceNumber;
        private String notes;
        private String imageUrl;
        private LocalDateTime paidAt;
    }
}

File: menu-scanner-frontend-owner/src/redux/features/subscription/store/models/response/subscription-history-response.ts
import { Pagination } from "@/utils/common/pagination";

export interface SubscriptionHistoryPaymentItem {
  paymentId: string;
  amount: number;
  paymentMethod: string;
  paymentType: string;
  status: string;
  referenceNumber: string | null;
  imageUrl: string | null;
  paidAt: string;
}

export interface SubscriptionHistoryResponseModel {
  subscriptionId: string;
  businessId: string;
  businessName: string;
  logoBusinessUrl?: string;
  planId: string;
  planName: string;
  planPrice: number;
  planDurationType: string;
  startDate: string;
  endDate: string;
  autoRenew: boolean;
  status: string;
  daysRemaining: number;
  paymentStatus: string;
  totalPaid: number;
  payment: SubscriptionHistoryPaymentItem | null;
}

export interface AllSubscriptionHistoryResponseModel extends Pagination {
  content: SubscriptionHistoryResponseModel[];
}
